public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -372,6 +380,57 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     return this._initialState;
   }
 
+  public duplicatePanel(vizPanel: VizPanel) {
```

Comment:
@harisrozajac - I think when duplicating a panel that is a child of `PanelRepeaterGridItem` we need to duplicate tha panel that lives in the `PanelRepeaterGridItem.state.source`. Or alternatively we need to make sure that the local variables that are representing the repeat context for the repeated panel, are removed from the state of the cloned one. Otherwise the interpolation in the panel will not work as expected. To notice the issue try duplicating the full-width panel in the dashbard I've earlier attached. You will notice that when selecting multiple values in the `query1` variable, the title of that panel will not be updated correctly.

I would suggest just going with the `PanelRepeaterGridItem.state.source` clone. This will keep us in parity with the current behavior, when cloning a repeated panel does not preserve the repeat config of such a panel.

Reply:
@dprokop - great catch! yeah, initially i didn't test this with the variables, but i've updated this and now it works as expected